tests/ttnn/unit_tests/operations/ccl/perf/test_all_gather_N300_post_commit.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import pytest
+import ttnn
+from models.utility_functions import skip_for_grayskull
+from tests.ttnn.unit_tests.operations.ccl.test_all_gather import (
+    run_all_gather_on_n300_impl,
+)
+
+
+# Enumerate the post-commit cases explicitly
+@skip_for_grayskull("Requires eth connected devices to run")
+@pytest.mark.parametrize(
+    "num_devices, num_links, output_shape, dim, layout",
+    [
+        (2, 1, [1, 1, 64, 16384], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [8, 5, 32, 768], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [1, 1, 32, 736], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [1, 1, 32, 704], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [1, 1, 64, 704], 3, ttnn.TILE_LAYOUT),
+        (2, 1, [1, 1, 32, 736], 3, ttnn.ROW_MAJOR_LAYOUT),
+        (2, 1, [1, 1, 32, 704], 3, ttnn.ROW_MAJOR_LAYOUT),
+        # (2, 1, [1, 1, 64, 704], 3, ttnn.ROW_MAJOR_LAYOUT),
```

Comment:
Can we remove these and other commented out cases? We can cleanup/choose specific cases after.